Sources/StreamChat/Workers/ChannelUpdater.swift

```diff
@@ -220,4 +220,34 @@ class ChannelUpdater<ExtraData: ExtraDataTypes>: Worker {
             completion?($0.error)
         }
     }
+    
+    /// Queries the watchers of a channel.
+    ///
+    /// For more information about channel watchers, please check [documentation](https://getstream.io/chat/docs/ios/watch_channel/?language=swift)
+    ///
+    /// - Parameters:
+    ///   - query: Query object for watchers. See `ChannelWatcherListQuery`
+    ///   - completion: Called when the API call is finished. Called with `Error` if the remote update fails.
+    func channelWatchers(query: ChannelWatcherListQuery, completion: ((Error?) -> Void)? = nil) {
+        apiClient.request(endpoint: .channelWatchers(query: query)) { (result: Result<ChannelPayload<ExtraData>, Error>) in
+            do {
+                let payload = try result.get()
+                self.database.write { (session) in
+                    if query.pagination.offset == nil || query.pagination.offset == 0 {
+                        // This is the first page of the watchers.
+                        // We can clear the existing watchers safely
+                        let channel = session.channel(cid: query.cid)
+                        channel?.watchers.removeAll()
```

Comment:
Let's do this only if the watchers from the payload are empty, too

Reply:
done